reward/staking_manager.go

```diff
@@ -109,36 +144,51 @@ func (sm *StakingManager) updateStakingInfo(blockNum uint64) (*StakingInfo, erro
 }
 
 // Subscribe setups a channel to listen chain head event and starts a goroutine to update staking cache.
-func (sm *StakingManager) Subscribe() {
-	sm.chainHeadSub = sm.blockchain.SubscribeChainHeadEvent(sm.chainHeadChan)
+func SubscribeStakingManager() {
+	if stakingManager == nil {
+		logger.Warn("unable to subscribe; stakingManager is not set; this can slow down node")
```

Comment:
Could you define and use an error constant variable for this case which `stakingManager` is nill case?

Reply:
review reflected